### PR TITLE
AVX-19373: add DiffSuppressFunc to tgw_peering and tgw_peering_domain_conn

### PR DESCRIPTION
**Pull request by shuoliugit**

### CODE REVIEW

File: aviatrix/resource_aviatrix_aws_tgw_peering_domain_conn.go [hunk 1]
```diff
@@ -20,27 +20,55 @@ func resourceAviatrixAWSTgwPeeringDomainConn() *schema.Resource {
...
-				ForceNew:    true,
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					tgwName2Old, _ := d.GetChange("tgw_name2")
+					domainName1Old, _ := d.GetChange("domain_name1")
+					domainName2Old, _ := d.GetChange("domain_name2")
+					return old == d.Get("tgw_name2").(string) && new == tgwName2Old.(string) &&
+						d.Get("domain_name1") == domainName2Old.(string) && d.Get("domain_name2") == domainName1Old.(string)
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_peering_domain_conn.go]: Is the logic here correct?

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

File: aviatrix/resource_aviatrix_aws_tgw_peering_domain_conn.go (hunk 1 above)
Comment (shuoliugit) [reply aviatrix/resource_aviatrix_aws_tgw_peering_domain_conn.go]: The logic here is that there should not be diff in this situation:

> new tgw1 == old tgw2
> new tgw2 == old tgw1
> new domain1 == old domain2
> new domain2 == old domain1

Comment (zhiqiangjin) [reply aviatrix/resource_aviatrix_aws_tgw_peering_domain_conn.go]: This is too ugly. Please create a function for it and move it to aws_tgw_peering_domain_conn.go or wherever is appropriate.

**Review by zhiqiangjin — CHANGES_REQUESTED** [review]

File: goaviatrix/aws_tgw_peering.go [hunk 2]
```diff
@@ -67,3 +69,13 @@ func (c *Client) DeleteAwsTgwPeering(awsTgwPeering *AwsTgwPeering) error {
 	awsTgwPeering.Async = true
 	return c.PostAsyncAPI(awsTgwPeering.Action, awsTgwPeering, BasicCheck)
 }
+
+func DiffSuppressFuncAwsTgwPeeringTgwName1(k, old, new string, d *schema.ResourceData) bool {
+	tgwName2Old, _ := d.GetChange("tgw_name2")
```
Comment (zhiqiangjin) [inline goaviatrix/aws_tgw_peering.go:74]: GetChange is not correct, it maybe

File: aviatrix/resource_aviatrix_aws_tgw_peering.go [hunk 3]
```diff
@@ -20,16 +20,18 @@ func resourceAviatrixAWSTgwPeering() *schema.Resource {
 
 		Schema: map[string]*schema.Schema{
 			"tgw_name1": {
-				Type:        schema.TypeString,
-				Required:    true,
-				ForceNew:    true,
-				Description: "Name of the first AWS tgw to make a peer pair.",
+				Type:             schema.TypeString,
+				Required:         true,
```
Comment (zhiqiangjin) [inline aviatrix/resource_aviatrix_aws_tgw_peering.go:24]: adsdgf

**Review by zhiqiangjin — APPROVED** [review]

**Pull request merged**